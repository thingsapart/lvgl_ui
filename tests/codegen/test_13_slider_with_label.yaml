# UI Test: A slider to control a value, with a label positioned above it.
- type: slider
  id: '@brightness_slider'
  width: 200
  center: []
  lv_slider_set_value: [80, LV_ANIM_ON]

- type: label
  align_to: ['@brightness_slider', LV_ALIGN_OUT_TOP_MID, 0, -15]
  text: Brightness
