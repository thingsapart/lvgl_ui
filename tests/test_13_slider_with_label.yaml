# UI Test: A slider to control a value, with a label positioned above it.
- type: slider
  id: '@brightness_slider'
  width: 200
  center: []
  value: 80

- type: label
  align_to: ['@brightness_slider', LV_ALIGN_OUT_TOP_MID, 0, -15]
  text: Brightness
