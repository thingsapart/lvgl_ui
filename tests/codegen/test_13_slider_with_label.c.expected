/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // brightness_slider: brightness_slider_0 (slider)
    lv_obj_t* brightness_slider_0 = lv_slider_create(parent);

    obj_registry_add("brightness_slider", brightness_slider_0);
    lv_obj_set_style_width(brightness_slider_0, 200, 0);
    lv_obj_center(brightness_slider_0);
    lv_slider_set_value(brightness_slider_0, 80, 1);

    // unnamed: label_1 (label)
    lv_obj_t* label_1 = lv_label_create(parent);

    lv_obj_align_to(label_1, brightness_slider_0, LV_ALIGN_OUT_TOP_MID, 0, -15);
    lv_label_set_text(label_1, "Brightness");

}
